services/UserAuthService.ts

```diff
@@ -28,10 +29,17 @@ export default class UserAuthService {
       const userRepository = Repositories.user(txn);
       const emailAlreadyUsed = !!(await userRepository.findByEmail(registration.email));
       if (emailAlreadyUsed) throw new BadRequestError('Email already in use');
+      if (registration.handle) {
+        const userHandleTaken = await userRepository.isHandleTaken(registration.handle);
+        if (userHandleTaken) throw new BadRequestError('This handle is already in use.');
+      }
+      const userHandle = registration?.handle
+         ?? UserAccountService.generateDefaultHandle(registration.firstName, registration.lastName);
```

Comment:
Do we need the optional chaining here? I'm not sure there's ever a case where `registration` is null or undefined. If we don't need it, let's remove it